libavoidrepetition.h

```diff
@@ -158,3 +158,49 @@ double min(double numbers[], int arrlen) {
 	}
 	return minitem;
 }
+
+
+
+size_t str_split(char ***array, char *str, const char *del) {
+    char *token;
+    size_t i = 0;
+
+    token = strtok_r(str, del);
+  
+    while(token != NULL) {
+        *array = realloc(*array, sizeof(char *) * (i + 1));
```

Comment:
Hey Arin @koreaneggroll, we are **libavoidrepetition**. We strive at reusing code, and just avoid repetition.

Don't you think, in C, we always need to `realloc()` whenever we just want to add/concat strings? What about we make a new function `add(char** destinationStr, char* addingStr)`? What do you say?

Reply:
I don't think I fully grasp what you mean by that, could you go more in detail ?

Reply:
Uhh... Think of like:
`add(the_main_string_variable_name_here, the_string_that_will_be_added_to_main_string);`

Simple enough?

Reply:
So like, take the string that was split and copy that value to another variable?

Reply:
For anything, which adds two strings. Consider thinking of this example:
```C
char* httpResponse = (char*)malloc(1024);
char* httpHeader = (char*)malloc(1024);
char* httpBody = (char*)malloc(1024);

/* --code that gets some http header-- */
str_add(httpResponse, httpHeader);
/* --code that gets some response body(HTML)-- */
str_add(httpResponse, httpBody);
```

Reply:
Isn't that just strcat ?

Reply:
Sure, it is -- unless we do the `malloc_reusable_size()` & `strlen()` checks, and `realloc()` accordingly. 

Sure we could just do:
`static char* httpResponse[100000];`

But I got injuriously ridiculed by a coder from reddit for using `[100000]`. And by using `static`, you agree to disallow operations on the string for future. So, seems like `malloc()` and `realloc()` is the only best practice.

Reply:
So you want to make a function that adds 2 strings togheter, and then make a third string and reallocate it memory and if it isn't big enough you reallocate it ? makes sense

Reply:
Yes. The plan is:
`realloc()` returns an entire new pointer alongwith the string up to which is concatenated. So, we could: 
```C
char* addingStrTemp = (char*)realloc(...);
```
 and use that. And finally: 
```C
    ...
    ...
    return addingStrTemp;
} //add(strDest, strToAdd)
```

Reply:
well you don't need a function for that, even writing it from scratch is pretty simple ```c
    char *str1 = malloc(sizeof(char) * 1024);
    char *str2 = malloc(sizeof(char) * 1024);

    strcpy(str1, "whatever");
    strcpy(str2, " whatever once again");

    char *str3 = malloc(sizeof(str1) + sizeof(str2));

    strcat(str1, str2);
    strcpy(str3, str1);```

Reply:
don't know why that didn't put the code corectly inside but meh

Reply:
Do we go through add these `strcpy` hassle and  dynamic `realloc` hassle every time we wanna add 2 strings?

No. We don't. So, let's hide that with an abstraction `add()` that just adds!

Reply:
alright then, ima do it